core/app/components/FormBuilder/ElementPanel/InputComponentConfigurationForm.tsx

```diff
@@ -136,7 +220,15 @@ export const InputComponentConfigurationForm = ({ index }: Props) => {
 	const component = form.watch("component");
 
 	const onSubmit = (values: ConfigFormData) => {
-		update(index, { ...selectedElement, ...values, updated: true, configured: true });
+		// Some `config` schemas have extra values which persist if we don't Clean first
+		const cleanedConfig = Value.Clean(componentConfigSchemas[values.component], values.config);
```

Comment:
for example, the RadioGroup has a field `includeOther` which the SelectDropdown does not have. without this line, if we had saved a component as a RadioGroup, then switched it to a SelectDropdown and saved that, the `includeOther` would still persist in `config`, when really we'd like `config` to look like what we expect of a `SelectDropdown` config